pkg/operator/bootstrap.go

```diff
@@ -22,10 +27,115 @@ import (
 // was specified.
 const randomSecretSize = 64
 
-func (h *Handler) Bootstrap() (*regopapi.ImageRegistry, error) {
-	crList := &regopapi.ImageRegistryList{
+func resourceName(namespace string) string {
+	if namespace == "default" {
+		return "docker-registry"
+	}
+	return "image-registry"
+}
+
+func newSpecFromDeploymentConfig(dc *appsapi.DeploymentConfig) (imageregistryapi.ImageRegistrySpec, error) {
+	fail := func(format string, a ...interface{}) (imageregistryapi.ImageRegistrySpec, error) {
+		format = "unable to create an image registry spec from the DeploymentConfig %s/%s: " + format
+		a = append([]interface{}{dc.Namespace, dc.Name}, a...)
+		return imageregistryapi.ImageRegistrySpec{}, fmt.Errorf(format, a...)
+	}
+
+	containers := dc.Spec.Template.Spec.Containers
+	if len(containers) != 1 {
```

Comment:
This may fail if the cluster has initialisation hooks that add sidecar containers for logging or something like this.